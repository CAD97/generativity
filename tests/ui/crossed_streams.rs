use generativity::make_guard;

fn main() {
    make_guard!(a);
    make_guard!(b);
    dbg!(a == b); // ERROR (here == is a static check)
}
