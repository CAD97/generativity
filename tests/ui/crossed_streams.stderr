error[E0597]: `tag` does not live long enough
 --> $DIR/crossed_streams.rs:5:5
  |
5 |     make_guard!(b);
  |     ^^^^^^^^^^^^^^ borrowed value does not live long enough
6 |     dbg!(a == b); // ERROR (here == is a static check)
7 | }
  | -
  | |
  | `tag` dropped here while still borrowed
  | borrow might be used here, when `_guard` is dropped and runs the `Drop` code for type `main::make_guard`
  |
  = note: values in a scope are dropped in the opposite order they are defined
  = note: this error originates in the macro `make_guard` (in Nightly builds, run with -Z macro-backtrace for more info)
